### DISCUSSION TITLE
Question about Euphony.js string length issue

### DISCUSSION THREAD

**Discussion by DahyeonWoo** [euphony-io/euphony#155] [Q&A] [answered]
Hello, I heard that there is a problem that data cannot be transmitted if the string of data is longer than a certain length when the speaker communicates with sound waves in the Euphony.js project.
I wonder how long the string can be sent.

**Comment by designe** [comment, marked as answer]
Hi! @defwdahyun0 
Thanks for your quesiton!

To be precise, the number of string values that can be sent by the transmitter is infinite. However, the problem is that noise is mixed in from the receiver stage that recognizes this, and error data is received, and it is not possible to process it as correct data.

The test below encodes and decodes data without any noise. 
This is fine even when testing very long data. because it is clean data without noise.
https://github.com/euphony-io/euphony/blob/master/euphony/src/main/cpp/tests/FSKTest.cpp

So, frankly this should be improved.
Now `euphony` is using infinite payload data in packet. 
In the near future, euphony should provide fixed finite data using multi packets.

Is it answer for your question?

**Reply by designe** [reply]
Same with euphony.js :)
